src/credentials/DeviceAttestationVerifier.cpp

```diff
@@ -52,6 +52,21 @@ class UnimplementedDACVerifier : public DeviceAttestationVerifier
         (void) certDeclBuffer;
         return AttestationVerificationResult::kNotImplemented;
     }
+
+    AttestationVerificationResult ValidateCertificateDeclarationPayload(const ByteSpan & certDeclBuffer,
+                                                                        const ByteSpan & firmwareInfo, uint16_t clusterVendorId,
```

Comment:
Since there is a large number of params, and they will need to be passed down from callers and passed into device attestation verification, I strongly suggest the following struct to be used, which will make the method signature more future-proof as well:

```
struct DeviceInfoForAttestation 
{
    // Vendor ID reported by device in Basic Information cluster
    uint16_t vendorId = 0;
    // Product ID reported by device in Basic Information cluster
    uint16_t productId = 0;
    // Vendor ID from  DAC
    uint16_t dacVendorId = 0;
    // Product ID from DAC
    uint16_t dacProductId = 0;
    // Vendor ID from  PAI cert
    uint16_t paiVendorId = 0;
    // Product ID from PAI cert (0 if absent)
    uint16_t paiProductId = 0;
    // Vendor ID from  PAA cert (0 if absent)
    uint16_t paaVendorId = 0;
};
```

This would also make passing the info to `VerifyAttestationInformation` much easier.

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/11838/commits/d0f7bb179727ea267192763cfd372584a9e7bd30